Source/Particles/WarpXParticleContainer.cpp

```diff
@@ -249,28 +249,64 @@ WarpXParticleContainer::DepositCurrent(WarpXParIter& pti,
 
     // Number of guard cells for local deposition of J
     WarpX& warpx = WarpX::GetInstance();
-    const int ng_J = warpx.get_ng_depos_J().max();
-
-    // Extract deposition order (same order along all directions) and check that
-    // particles shape fits within the guard cells.
-    // NOTE: In specific situations where the staggering of J and the current
-    // deposition algorithm are not trivial, this check might be too relaxed
-    // and we might include a particle that should deposit part of its current
-    // in a neighboring box. However, this should catch particles traveling many
-    // cells away, for example with algorithms that allow for large time steps.
-    const int shape_extent = static_cast<int>(WarpX::nox / 2);
+
+    const amrex::IntVect& ng_J = warpx.get_ng_depos_J();
+
+    // Extract deposition order and check that particles shape fits within the guard cells.
+    // NOTE: In specific situations where the staggering of J and the current deposition algorithm
+    // are not trivial, this check might be too relaxed and we might include a particle that should
+    // deposit part of its current in a neighboring box. However, this should catch particles
+    // traveling many cells away, for example with algorithms that allow for large time steps.
+
+    // Along x
+    int shape_extent = static_cast<int>(WarpX::nox / 2);
+#ifndef AMREX_USE_GPU
+    // On CPU: Particles deposit on tile arrays,
+    // which have a small number of guard cells ng_J
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
+        amrex::numParticlesOutOfRange(pti, ng_J[0] - shape_extent) == 0,
```

Comment:
I think that this call to `numParticlesOutOfRange` will check that particles are within `ng_J[0]` in **all dimensions**, not just in `x`. This might have to be changed.

Reply:
As discussed off-line, this might need a version of `numParticlesOutOfRange` that takes an `IntVect` as input.

Reply:
@RemiLehe @atmyers Thank you both, will use https://github.com/AMReX-Codes/amrex/pull/1751/ as soon as it is ready.